### PR TITLE
Remove old branch plugin tests

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins/foreman_discovery.yaml [hunk 1]
```diff
@@ -4,9 +4,5 @@
     branch:
       - develop:
           foreman_branch: develop
-      - 15.0-stable:
-          foreman_branch: 1.22-stable
-      - 15.1-stable:
-          foreman_branch: 1.23-stable
```
Comment (mmoll) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins/foreman_discovery.yaml]: @lzap Should new versions get added here?

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins/foreman-tasks.yaml [hunk 2]
```diff
@@ -4,9 +4,5 @@
     branch:
       - master:
           foreman_branch: develop
-      - 0.16.x:
-          foreman_branch: 1.23-stable
-      - 0.15.x:
-          foreman_branch: 1.22-stable
```
Comment (mmoll) [inline puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins/foreman-tasks.yaml]: @adamruzicka Should new versions get added here?

**Comment by ekohl** [comment]
Until 2.0 goes GA, 1.23 is supported. Should we keep those versions around for a bit longer?

**Comment by mmoll** [comment]
I'm unsure if anybody cares about the results, tbh :)

File: puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins/foreman-tasks.yaml (hunk 2 above)
Comment (adamruzicka) [reply puppet/modules/jenkins_job_builder/files/theforeman.org/yaml/jobs/plugins/foreman-tasks.yaml]: That would make sense to me

**Pull request merged**